src/AI/InfantrySpamAI.java

```diff
@@ -17,21 +23,173 @@
 public class InfantrySpamAI implements AIController
 {
   Queue<GameAction> actions = new ArrayDeque<GameAction>();
-  
+
   private Commander myCo = null;
-  
+
+  private ArrayList<XYCoord> unownedProperties;
+  private ArrayList<XYCoord> capturingProperties;
+
+  private StringBuffer logger = new StringBuffer();
+  private int turnNum = 0;
+
   public InfantrySpamAI(Commander co)
   {
     myCo = co;
   }
 
   @Override
-  public void initialize(GameMap gameMap)
-  { /* No initialization needed */ }
+  public void initTurn(GameMap gameMap)
+  {
+    turnNum++;
+    log(String.format("[======== ISAI initializing turn %s for %s =========]", turnNum, myCo));
+
+    // Make sure we don't have any hang-ons from last time.
+    actions.clear();
+    
+    // Create a list of every property we don't own, but want to.
+    unownedProperties = new ArrayList<XYCoord>();
+    for( int x = 0; x < gameMap.mapWidth; ++x )
+    {
+      for( int y = 0; y < gameMap.mapHeight; ++y )
+      {
+        XYCoord loc = new XYCoord(x, y);
+        if( gameMap.getLocation(loc).isCaptureable() && myCo.isEnemy(gameMap.getLocation(loc).getOwner()) )
+        {
+          unownedProperties.add(loc);
+        }
+      }
+    }
+    capturingProperties = new ArrayList<XYCoord>();
+    for( Unit unit : myCo.units )
+    {
+      if( unit.getCaptureProgress() > 0 )
+      {
+        capturingProperties.add(unit.getCaptureTargetCoords());
+      }
+    }
+
+    // If the CO has enough AP, preload the CommanderAbilityAction.
+    ArrayList<CommanderAbility> abilities = myCo.getReadyAbilities();
+    if( abilities.size() > 0 )
+    {
+      actions.offer(new GameAction.AbilityAction(abilities.get(0)));
+    }
+  }
+
+  @Override
+  public void endTurn()
+  {
+    log(String.format("[======== ISAI ending turn %s for %s =========]", turnNum, myCo));
+    System.out.println(logger.toString());
+    logger = new StringBuffer();
+  }
+
+  private void log(String message)
+  {
+    System.out.println(message);
+    logger.append(message).append('\n');
+  }
```

Comment:
What's this for?
Looks like you print the message both when you receive it *and* at the end of the turn, but it only gets printed once.

Reply:
Nah, it is printing it twice. I was switching between the two while debugging, and left both enabled.